src/server/middleware/readonly.go

```diff
@@ -1,27 +1,55 @@
 package middleware
 
 import (
+	"net/http"
+
 	"github.com/goharbor/harbor/src/common/utils/log"
 	"github.com/goharbor/harbor/src/core/config"
 	internal_errors "github.com/goharbor/harbor/src/internal/error"
-	"net/http"
 )
 
-type readonlyHandler struct {
-	next http.Handler
+// ReadOnlyConfig defines the config for ReadOnly middleware.
+type ReadOnlyConfig struct {
```

Comment:
Similar Q with above ORM middleware?

Reply:
Yes, `ReadOnly func(*http.Request)` as configuration, so that this middleware can be used in many different scenarios and also make it more easy to test.